src/eva/plot_tools/dynamic_config.py

```diff
@@ -0,0 +1,132 @@
+# (C) Copyright 2021-2022 NOAA/NWS/EMC
+#
+# (C) Copyright 2021-2022 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration. All Rights Reserved.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+import math
+import numpy as np
+from scipy.stats import skew
+
+from eva.utilities.utils import replace_vars_dict
+
+
+# --------------------------------------------------------------------------------------------------
+
+
+def vminvmaxcmap(logger, option_dict, plots_dict, data_collections):
+
+    # Get percentage of data to use in computing limits. Code will sort the data
+    # by size and then trim the minimum and maximum until this percentage of data
+    # is kept
+    percentage_capture = option_dict.get('percentage capture', 100)
+
+    # Optionally the data might have a channel.
+    channel = option_dict.get('channel', None)
+
+    # Get the data variable to use for determining cbar limits
+    varname = option_dict.get('data variable')
+    varname_cgv = varname.split('::')
+    datavar = data_collections.get_variable_data(varname_cgv[0], varname_cgv[1],
+                                                 varname_cgv[2], channel)
+
+    # Reorder the data array
+    datavar = np.sort(datavar)
+
+    # Decide how many values to throw out on each end of the dataset
+    n = datavar.size
+    n_throw_out = ((100-percentage_capture) * n / 100) / 2
+
+    # The value needs to be an integer
+    n_throw_out = np.floor(n_throw_out)
+
+    # Create new array with only the data to consider
+    if n_throw_out == 0:
+        datavar_check = datavar
+    else:
+        datavar_check = datavar[n_throw_out:-n_throw_out]
+
+    # Find minimum and maximum values
+    cmap = option_dict.get('sequential colormap', 'viridis')
+    vmax = np.nanmax(datavar_check)
+    vmin = np.nanmin(datavar_check)
+
+    # If positive and negative values are present then a diverging colormap centered on zero should
+    # be used.
+    if vmin < 0.0 and vmax > 0.0:
+        vmax = np.nanmax(np.abs(datavar_check))
```

Comment:
What if the max value is 2 and the min value is -50? We need to compute min and max and then get the max of the abs value of each to use as the colorbar endpoints.

Reply:
In that case wouldn't the result of `np.abs(datavar_check)` be 50? Then `vmax` would be 50 and `vmin` -50.

Reply:
ah, yeah, duh.